pkgs/development/tools/misc/pahole/threading-reproducibility.patch

```diff
@@ -0,0 +1,18 @@
+diff --git a/pahole.c b/pahole.c
+index 6fc4ed6..a4e306f 100644
+--- a/pahole.c
++++ b/pahole.c
+@@ -1687,8 +1687,11 @@ static error_t pahole__options_parser(int key, char *arg,
+ 		  class_name = arg;			break;
+ 	case 'j':
+ #if _ELFUTILS_PREREQ(0, 178)
+-		  conf_load.nr_jobs = arg ? atoi(arg) :
+-					    sysconf(_SC_NPROCESSORS_ONLN) * 1.1;
++		  // Default to 1 thread if reproducibility is desirable.
```

Comment:
that's stronger than default: you literally ignore any attempt at setting -j or -jX if SOURCE_DATE_EPOCH is set, this comment is misleading.

Reply:
Ah, indeed -- let me fix that, it wasn't intended :) thanks!

Reply:
(to clarify this: pahole already defaults to single thread as far as I can see; the `arg ?` ternary here is just argparse's way of saying "`-j` (without param) will default to no arg e.g. nproc*1.1 otherwise `-jX` will set to X")

Reply:
> pahole already defaults to single thread

That doesn't match my testing, so I wonder if something upstream in Linux is pass a -j flag. Or I got unlucky with my testing... Let me double check all that.

Reply:
https://github.com/torvalds/linux/blob/bb7c241fae6228e89c0286ffd6f249b3b0dea225/scripts/pahole-flags.sh#L20 ok, at least I'm not completely wrong...

Thoughts on just ignoring -j (keeping the current code, fixing the comment / commit message) when SOURCE_DATE_EPOCH is set?

We could also just patch Linux instead, pahole isn't really used by much else so in practice it's probably good enough -- but I think I have a weak preference for fixing that in pahole instead.

Reply:
We can ignore argless `-j` ? But to be honest SOURCE_DATE_EPOCH is a pretty good discriminant, I don't mind the current patch as long as comment is fixed

Reply:
Ignoring just argless isn't really enough, "-j N" would be equally unreproducible.